zepl/contrib/load_files.vim

```diff
@@ -4,11 +4,30 @@
 " Legal:        No rights reserved.  Public domain.
 
 function! zepl#contrib#load_files#load(...) abort
-    let fns = (a:0 > 0 ? a:000 : [expand('%')])
-    let fns = map(copy(fns), 'fnamemodify(expand(v:val), ":p")')
-    " let fns = filter(uniq(copy(fns)), 'filereadable(v:val)')
-    let cmds = map(copy(fns), 'printf(b:repl_config["load_file"], v:val)')
-    call zepl#send(cmds)
+    " The dictionary entry identifying the command to laod a file
+    let loader_cmd_varname = 'load_file'
+
+    " Check buffer-local config, then global config
+    if exists('b:repl_config') && has_key(b:repl_config, &filetype)
+      let config_entry = b:repl_config[&filetype]
+    elseif exists('g:repl_config') && has_key(g:repl_config, &filetype)
+      let config_entry = g:repl_config[&filetype]
+    else
+      echoerr 'Neither b:repl_config nor g:repl_config has a definition for the filetype "'.&filetype.'"'
+      return
+    endif
+
+    if has_key(config_entry, loader_cmd_varname)
+      let load_cmd = config_entry[loader_cmd_varname]
+
+      let fns = (a:0 > 0 ? a:000 : [expand('%')])
+      let fns = map(copy(fns), 'fnamemodify(expand(v:val), ":p")')
+      " let fns = filter(uniq(copy(fns)), 'filereadable(v:val)')
+      let cmds = map(copy(fns), 'printf(load_cmd, v:val)')
+      call zepl#send(cmds)
+    else
+      echoerr 'Neither b:repl_config nor g:repl_config has "load_file" defined.'
+    endif
```

Comment:
I've pushed a change which adds a `zepl#config` function so that contrib files can share the look up logic.

By using this new function you can simplify this code to the following:

```suggestion
    let load_cmd = zepl#config('load_file', '')

    if empty(load_cmd)
        echoerr "Could not load file(s) into REPL.  No value for 'load_file' set in Zepl configuration.  See ':help zepl-load_files' for more details."
    else
        let fns = (a:0 > 0 ? a:000 : [expand('%')])
        let fns = map(copy(fns), 'fnamemodify(expand(v:val), ":p")')
        " let fns = filter(uniq(copy(fns)), 'filereadable(v:val)')
        let cmds = map(copy(fns), 'printf(load_cmd, v:val)')
        call zepl#send(cmds)
    endif
```

Reply:
Nice! Done.